.github/workflows/publish.yml

```diff
@@ -58,10 +58,14 @@ jobs:
           print(f"::set-output name=publishing::{publishing}")
 
       - name: Set up QEMU (for docker buildx)
-        uses: docker/setup-qemu-action@v1
+        # NOTE: Updates to the actions commit hash should go hand in hand with
+        #       updates in the test-docker-build workflow.
+        uses: docker/setup-qemu-action@25f0500ff22e406f7191a2a8ba8cda16901ca018 # associated tag: v1.0.2
```

Comment:
Dependabot is already configured on this repo for github actions: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/eb4dd3b2becd33b3db0d2b09daafd9d66c26e970/.github/dependabot.yaml#L38-L47

Is the `# associated tag: v1.0.2` going to cause problems?

Reply:
I'm not sure, but i would guess not except that it may not update it with the annotation.

I think we shouldn't update these much in general and lean towards thinking we only do it when we see a need for it rather than when dependabot suggests it.

I suggest we wait and learn what happens and make a decision down the line about things in more detail at that point

Reply:
OK, would you mind adding a comment along those lines? I'm thinking about this fropm the perspective of a new or infrequent maintainer, there's nothing to say don't merge PRs by dependabot if they change this line.

Reply:
Hmmm, I'm a bit confused - could you push or create a code suggestion?

I don't see it as a problem if we end up with a `# associated tag: v1.0.2` being stripped if dependabot updates this line of code. Are you thinking about if it would retain the comment, and it now points to an outdated tag because dependabot managed to retain the version comment but update the hash?

- Confused about what to expect from dependabot when pinning a hash
- Confused about what your concern is out of several concerns I can imagine (comment outdated, stripped, action out of sync, perhaps some more)

Reply:
Typically dependabot updates are OK to merge if they're non-breaking and CI passes. You've made an alternative suggestion:
> I think we shouldn't update these much in general and lean towards thinking we only do it when we see a need for it rather than when dependabot suggests it.

But that's going to be lost as soon as this PR is merged.

Reply:
What policy do you prefer? Merge dependabot PRs as usual in which case this can be merged as is and if the comment causes a problem we can fix it later, or document a new policy of only merging when it's needed?

Reply:
@manics hmmm this became a bit complicated for me to grasp fully, so my answer below is vague because I'm a bit lost now.

I'm still uncertain about aspects like if dependabot will even try to bump a commit-pinned action, and if so, to what version it will bump. Whenever a new release is made, whenever a new commit is made, or never?

Since I don't know that, and I feel that the inline comment about keeping versions in sync is around as a warning that something could go wrong if they get out of sync, and I'm very uncertain it would risk becoming out of sync by having dependabot - i lean towards not taking any action.

With regards to a policy for bumping the version of the QEMU action (like "we shouldn't update the much in general" or "only when found to be needed" or "always keep updated!" or "when you have time to check for changes upstream" etc): I think I'm too vaguely opined to motivate we follow a certain policy.

Reply:
+1 for merge as is and +1 to approve/merge any suggested policy or change you think make sense - I myself feel too vaguely opined and weakly informed about how dependabot will work to suggest something concretely.

Reply:
I'm happy with the status quo of merging all updates :smile: